### PR TITLE
Refactor: Merge client and server `Request`

### PR DESCRIPTION
**Pull request by tusharmath**
Merging `Client.ClientRequest` and `Request` into one type. This was unnecessarily duplicated.

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/service/EncodeClientParams.scala [hunk 1]
```diff
@@ -1,41 +1,42 @@
...
+      val url    = req.url
+
+      // As per the spec, the path should contain only the relative path.
+      // Host and port information should be in the headers.
+      val path = url.relative.encode
+
+      //    val content = req.getBodyAsString match {
+      //      case Some(text) => Unpooled.copiedBuffer(text, HTTP_CHARSET)
+      //      case None       => Unpooled.EMPTY_BUFFER
+      //    }
```
Comment (smehta91) [inline zio-http/src/main/scala/zhttp/service/EncodeClientParams.scala]: Please remove unnecessary code.

Comment (tusharmath) [reply zio-http/src/main/scala/zhttp/service/EncodeClientParams.scala]: Thank you 🙏

**Review by smehta91 — APPROVED** [review]

**Review by girdharshubham — APPROVED** [review]

File: zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  )
+
+  val clientParamWithAbsoluteUrl = HttpGen.clientRequest(
+    dataGen = HttpGen.httpData(
+      Gen.listOf(Gen.alphaNumericString),
+    ),
+    urlGen = HttpGen.genAbsoluteURL,
+  )
+
+  def clientParamWithFiniteData(size: Int): Gen[Random with Sized, Request] = HttpGen.clientRequest(
```
Comment (girdharshubham) [inline zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala:25]: #nitpick  Should we keep it HttpGen?

File: zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import io.netty.handler.codec.http.{HttpHeaderNames, HttpVersion}
+import zhttp.internal.HttpGen
+import zhttp.service.EncodeClientParams
+import zio.random.Random
+import zio.test.Assertion._
+import zio.test._
+
+object EncodeRequestSpec extends DefaultRunnableSpec with EncodeClientParams {
+
+  val anyClientParam: Gen[Random with Sized, Request] = HttpGen.clientRequest(
```
Comment (girdharshubham) [inline zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala:12]: #nitpick  This as well?
candidate for HttpGen?

Comment (amitksingh1490) [reply zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala:12]: we can keep it here for now, as this test is the only user of this Gen

File: zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala (hunk 2 above)
Comment (amitksingh1490) [reply zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala:25]: we can keep it here for now, as this test is the only user of this Gen

File: zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala (hunk 3 above)
Comment (tusharmath) [reply zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala:12]: This was already there, the file has been renamed only.

File: zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala (hunk 2 above)
Comment (tusharmath) [reply zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala:25]: This was already there, the file has been renamed only.

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**